File: app/src/main/java/com/zhxh/codeproj/Log.java
package com.zhxh.codeproj;

public class Log {

    private static final String dot = " ";

    public static void e(String tag, String msg) {
        System.out.println(tag + dot + msg);
    }

    public static void d(String tag, String msg) {
        System.out.println(tag + dot + msg);
    }

    public static void i(String tag, String msg) {
        System.out.println(tag + dot + msg);
    }

    public static void v(String tag, String msg) {
        System.out.println(tag + dot + msg);
    }


    public static void e(String msg) {
        System.out.println(msg);
    }

    public static void d(String msg) {
        System.out.println(msg);
    }

    public static void i(String msg) {
        System.out.println(msg);
    }

    public static void v(String msg) {
        System.out.println(msg);
    }

}

File: app/src/main/java/com/zhxh/codeproj/kotlin/examples/util/log.kt
package com.zhxh.codeproj.kotlin.examples.util

import android.os.Build
import android.support.annotation.RequiresApi
import java.time.*

@RequiresApi(Build.VERSION_CODES.O)
fun log(msg: String) = println("${Instant.now()} [${Thread.currentThread().name}] $msg")